File: 3550. Smallest Index With Digit Sum Equal to Index.java
class Solution {
    public int smallestIndex(int[] nums) {
        ArrayList<Integer> list=new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            int temp=0;
            int n=nums[i];
            while (n>0){
                int r=n%10;
                n=n/10;
                temp+=r;
            }
            if (i==temp){
                list.add(i);
            }
        }
       
        Collections.sort(list);
        return list.isEmpty()?-1:list.get(0);
    }
}

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        small=10000
        for i in range(len(nums)):
            s=str(nums[i])
            c=0
            for j in range(len(s)):
                c+=int(s[j])
            if i==c:
                if small>nums[i]:
                    return i
        return -1



        